### PR TITLE
OrderTXO proposal

### PR DESCRIPTION
**Pull request by shogochiai**

### CODE REVIEW

**Comment by syuhei176** [comment]
@shogochiai You don't need to remove SimpleToken.
SimpleToken is not for DEX. just to generate the standard contract.
And it needs for unit-test of the compiler.
Please add just OrderToken.chr

https://github.com/cryptoeconomicslab/plasma-chamber/blob/master/packages/chamber-rootchain/contracts/TxVerification.sol#L45

File: example/OrderToken.chr [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  @beforeFilter onlyPair()
+  pub clause exchange(TXO txo) {
+    var owners = this.owners
+    this.owners = txo.owners
+    txo.owners = owners
+    this.state = null
+    
+    // Transaction Outputs of this Tx
+    return [
+      this,
```
Comment (syuhei176) [inline example/OrderToken.chr]: This is an interesting point.
We have 2 expressions.

```
return [txo]
```
or
```
return [this, txo]
```

I think `this` is always in outputs, so we need not to write manually.

Comment (shogochiai) [reply example/OrderToken.chr]: As design philosophy, I believe at least these items must be `grammatically-explicit` and `type-checked`.

- `Transaction Input (Transaction Output)`
- `Args`
- `Transaction Output`
  - `pwners`
  - `values`
  - `state`
  - `blknum`

This is why I remained `this` in `return` syntax, could you understand it?
And yes, `return this` is somehow unsexy syntax. We can seek for better syntax with the same purpose.

Comment (shogochiai) [reply example/OrderToken.chr]: Keep putting stateful TXO as `this`.

This is very unique design perspective.

Maybe, this statefle TXO is, for some sense NOT a TXO anymore.

More like, this stateful TXO is the Contract itself.

Thus, to assign that TXO to `this` is quiet logical.

`this = contract = stateful TXO`
very intuitive equation.

But first-time-developer wouldn't guess why it is designated so. We must design some guideline.

Comment (shogochiai) [reply example/OrderToken.chr]: Summarized insight.
https://speakerdeck.com/shogochiai/stateful-txo-is-the-new-contract

**Comment by shogochiai** [comment]
I made current `SellOrder.chr`'s peg file/test.
Next I want to add

1. Lifecycle abstraction
2. Exit safety abstraction
3. argValues&argData abstraction

**Comment by shogochiai** [comment]
Added anotators for the after-exit-workaround.

`@refundable` appends `refund()` clause to a contract, then force execute it on rootchain after the statefulTXO is to be withdrawn. Most of clauses and most of constructor would be so.

`@continuable(Type arg)` annotator enable to continue a contract after withdrawal. Args are for limiting only the primitive state to be stored in rootchain because storage is expensive. Whole lifetime cycle of the Contract must be copied to rootchain in order to continue it. The beloved usage of this is 1) Big bet gambling's very last timing(Operator can force stop the battle) 2) 2rd layer Plasma contract for 3rd layer childchain (Put the contract on 1st layer as [Actor model contract](https://ethresear.ch/t/ethereum-2-0-data-model-actors-and-assets/4117))

**Pull request merged**